pkg/monitor/monitor/util.go

```diff
@@ -370,6 +370,12 @@ chmod 777 /data/prometheus /data/grafana
 			})
 
 	}
+	for k, v := range monitor.Spec.Initializer.Envs {
+		container.Env = append(container.Env, core.EnvVar{
+			Name:  k,
+			Value: v,
+		})
+	}
```

Comment:
do you allow users to override builtin envs?

if not, you can use this utility function https://github.com/pingcap/tidb-operator/blob/9dbd0b7ccc3602405e3046e50f006bca3f7361df/pkg/util/util.go#L208-L219

Reply:
I copied this from the behavior used in the same file for monitor.Spec.Grafana.Envs. I don't know if either of the behaviors is good or proper, but at least they'll be consistent 🙂 https://github.com/pingcap/tidb-operator/blob/2d1e25d132b7171174bed837d7d0e4973f87a690/pkg/monitor/monitor/util.go#L514

Reply:
@kolbe With your code change, there will be two envs for `GF_TIDB_PROMETHEUS_URL` in the yaml with the case in the original issue, right?
Would you mind define the following function and apply it to all of the four containers in the Monitor Pod so that we do not need to update code again if someone else wants to set env for other containers later?

// AppendOverwriteEnv appends envs `b` into `a` and overwrites the envs whose names already exist
// in `b`.
// Note that this will not change relative order of envs.
func AppendOverwriteEnv(a []corev1.EnvVar, b []corev1.EnvVar) []corev1.EnvVar {